packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
+  render(<CounterMock value={5} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons.length).toBe(2);
+  expect(buttons).toHaveLength(2);
 });
 
 test('buttons are disabled when value hits max or min', () => {
-  const { getAllByRole, rerender } = render(counterMock({ ...requiredAttributes, value: 0 }));
+  const { rerender } = render(<CounterMock value={0} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons[0]).toHaveProperty('disabled', true);
-  expect(buttons[1]).toHaveProperty('disabled', false);
+  expect(buttons[0]).toBeDisabled;
+  expect(buttons[1]).toBeEnabled;
 
-  rerender(counterMock({ ...requiredAttributes, value: 10 }));
+  rerender(<CounterMock value={10} />);
 
-  expect(buttons[0]).toHaveProperty('disabled', false);
-  expect(buttons[1]).toHaveProperty('disabled', true);
+  expect(buttons[0]).toBeEnabled;
+  expect(buttons[1]).toBeDisabled;
 });
 
 test('value prop only accepts whole numbers', () => {
-  const { container } = render(counterMock({ max: 20, ...requiredAttributes }));
+  render(<CounterMock value={5} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
+  const fractionalValue = '1.7';
 
-  const input = container.getElementsByTagName('input');
-  fireEvent.change(input[0], { target: { value: 1.5 } });
+  userEvent.clear(input);
+  userEvent.type(input, fractionalValue);
 
-  expect(handleChange).toHaveBeenCalledWith(2);
+  expect(input).toHaveValue('17');
 });
 
 test('value increases when increase or decrease icons are clicked', () => {
-  const { getByDisplayValue, container } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5') as HTMLInputElement;
-
-  const icons = container.getElementsByTagName('svg');
+  const counterValue = screen.getByRole<HTMLInputElement>('textbox').value;
+  const icons = Array.from(screen.getAllByTitle(/[a-z]{2}crease count/i)).map(
+    (title: HTMLElement): HTMLElement | null => title.parentElement,
+  );
 
-  expect(counter.value).toEqual('5');
-  fireEvent.click(icons[1]);
+  expect(counterValue).toEqual('5');
+  icons[1] && userEvent.click(icons[1]);
 
   expect(handleChange).toHaveBeenCalledWith(6);
-  fireEvent.click(icons[0]);
+  icons[0] && userEvent.click(icons[0]);
   expect(handleChange).toHaveBeenCalledWith(4);
 });
 
 test('value increases and decreases with arrow keypresses', () => {
-  const { getByDisplayValue } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5') as HTMLInputElement;
+  const counter = screen.getByRole<HTMLInputElement>('textbox');
```

Comment:
I don't think the type (`HTMLInputElement`) is necessary anywhere in the file, consider removing it.